File: core/src/main/java/y2k/joyreactor/common/PartialResult.java
package y2k.joyreactor.common;

/**
 * Created by y2k on 08/12/15.
 */
public class PartialResult<T> {

    public T result;

    public int progress;
    public int max;

    public static <T> PartialResult<T> complete(T data) {
        PartialResult<T> result = new PartialResult<>();
        result.result = data;
        return result;
    }

    public static <T> PartialResult<T> inProgress(int progress, int max) {
        PartialResult<T> result = new PartialResult<>();
        result.progress = progress;
        result.max = max;
        return result;
    }
}

File: core/src/main/kotlin/y2k/joyreactor/common/PartialResult.kt
package y2k.joyreactor.common

/**
 * Created by y2k on 08/12/15.
 */
data class PartialResult<T>(val result: T?, val progress: Int, val max: Int) {

    companion object {

        fun <T> complete(data: T): PartialResult<T> {
            return PartialResult(data, 0, 0)
        }

        fun <T> inProgress(progress: Int, max: Int = 100): PartialResult<T> {
            return PartialResult(null, progress, max)
        }
    }
}